src/EDMF_functions.jl

```diff
@@ -516,26 +530,30 @@ function compute_up_tendencies!(edmf::EDMFModel, grid::Grid, state::State, param
         q_tot_up = aux_up_i.q_tot
         q_tot_en = aux_en.q_tot
         θ_liq_ice_en = aux_en.θ_liq_ice
+        e_tot_en = aux_en.e_tot
+        h_tot_en = aux_en.h_tot
         θ_liq_ice_up = aux_up_i.θ_liq_ice
+        e_tot_up = aux_up_i.e_tot
+        h_tot_up = aux_up_i.h_tot
         entr_turb_dyn = aux_up_i.entr_turb_dyn
         detr_turb_dyn = aux_up_i.detr_turb_dyn
-        θ_liq_ice_tendency_precip_formation = aux_up_i.θ_liq_ice_tendency_precip_formation
+        e_tot_tendency_precip_formation = aux_up_i.e_tot_tendency_precip_formation
         qt_tendency_precip_formation = aux_up_i.qt_tendency_precip_formation
 
         ρarea = prog_up[i].ρarea
-        ρaθ_liq_ice = prog_up[i].ρaθ_liq_ice
+        ρae_tot = prog_up[i].ρae_tot
         ρaq_tot = prog_up[i].ρaq_tot
 
         tends_ρarea = tendencies_up[i].ρarea
-        tends_ρaθ_liq_ice = tendencies_up[i].ρaθ_liq_ice
+        tends_ρae_tot = tendencies_up[i].ρae_tot
         tends_ρaq_tot = tendencies_up[i].ρaq_tot
 
         @. tends_ρarea =
             -∇c(wvec(LBF(Ic(w_up) * ρarea))) + (ρarea * Ic(w_up) * entr_turb_dyn) - (ρarea * Ic(w_up) * detr_turb_dyn)
 
-        @. tends_ρaθ_liq_ice =
-            -∇c(wvec(LBF(Ic(w_up) * ρaθ_liq_ice))) + (ρarea * Ic(w_up) * entr_turb_dyn * θ_liq_ice_en) -
-            (ρaθ_liq_ice * Ic(w_up) * detr_turb_dyn) + (ρ_c * θ_liq_ice_tendency_precip_formation)
+        @. tends_ρae_tot =
+            -∇c(wvec(LBF(Ic(w_up) * ρarea * h_tot_up))) +  ρarea * Ic(w_up) * (entr_turb_dyn * h_tot_en - detr_turb_dyn * h_tot_up)
+            (ρarea * h_tot_up * Ic(w_up) * detr_turb_dyn) + (ρ_c * e_tot_tendency_precip_formation)
```

Comment:
```suggestion
            -∇c(wvec(LBF(Ic(w_up) * ρarea * h_tot_up))) +  ρarea * Ic(w_up) * (entr_turb_dyn * h_tot_en - detr_turb_dyn * h_tot_up) 
            + (ρ_c * e_tot_tendency_precip_formation)
```

I think you are repeating a source here.